### PR TITLE
1upocarina

### PR DESCRIPTION
**Pull request by ziptyze**
Addition of a new keyboard to the 1upkeyboards lineup

## Description

Adding the following keyboards:

1upocarina - a 5 button hot swap keyboard designed for playing osu

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/1upkeyboards/1upslider8/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                                   KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [9] = LAYOUT_2x4(
+                                   KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
```
Comment (zvecr) [inline keyboards/1upkeyboards/1upslider8/keymaps/default/keymap.c:80]:
```suggestion
    )
```

File: keyboards/1upkeyboards/1upslider8/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    [0] =  { ENCODER_CCW_CW(RGB_MOD,  RGB_RMOD)},
+    [1] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [2] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [3] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [4] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [5] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [6] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [7] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [8] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [9] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)}
```
Comment (zvecr) [inline keyboards/1upkeyboards/1upslider8/keymaps/default/keymap.c:94]:
```suggestion
    [0] =  { ENCODER_CCW_CW(RGB_MOD,  RGB_RMOD)}
```

File: keyboards/1upkeyboards/1upslider8/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    [0] =  { ENCODER_CCW_CW(RGB_MOD,  RGB_RMOD)},
+    [1] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [2] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [3] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [4] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [5] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [6] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [7] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [8] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)},
+    [9] =  { ENCODER_CCW_CW(KC_TRNS,  KC_TRNS)}
```
Comment (zvecr) [inline keyboards/1upkeyboards/1upslider8/keymaps/via/keymap.c:94]:
```suggestion
    [0] =  { ENCODER_CCW_CW(RGB_MOD,  RGB_RMOD)}
```

File: keyboards/1upkeyboards/1uposu5k/readme.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# 1uposu5k
+
+*A short description of the keyboard/project*
```
Comment (zvecr) [inline keyboards/1upkeyboards/1uposu5k/readme.md:3]: Should be filled our

File: keyboards/1upkeyboards/1upslider8/readme.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+* Keyboard Maintainer: [ziptyze](https://github.com/ziptyze)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make 1upslider8:default
+
+Flashing example for this keyboard:
+
+    make 1upslider8:default:flash
```
Comment (zvecr) [inline keyboards/1upkeyboards/1upslider8/readme.md:13]:
```suggestion
    make 1upkeyboards/1upslider8:default

Flashing example for this keyboard:

    make 1upkeyboards/1upslider8:default:flash
```

File: keyboards/1upkeyboards/1upslider8/readme.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+This keyboard is the slider8 from 1upkeyboards.com, a small macropad with an oled, slide potentiometer, rotary encoder, and 8 switches in a 2x4 ortholinear arrangement with per-key in-switch rgb leds.
```
Comment (zvecr) [inline keyboards/1upkeyboards/1upslider8/readme.md:1]:
```suggestion
# slider8

This keyboard is the slider8 from 1upkeyboards.com, a small macropad with an oled, slide potentiometer, rotary encoder, and 8 switches in a 2x4 ortholinear arrangement with per-key in-switch rgb leds.
```

File: keyboards/1upkeyboards/1uposu5k/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [8] = LAYOUT_1x5(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [9] = LAYOUT_1x5(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
```
Comment (zvecr) [inline keyboards/1upkeyboards/1uposu5k/keymaps/via/keymap.c:58]:
```suggestion
    )
```

File: keyboards/1upkeyboards/1uposu5k/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
+    [1] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [2] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [3] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [4] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [5] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [6] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [7] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [8] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [9] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)}
```
Comment (zvecr) [inline keyboards/1upkeyboards/1uposu5k/keymaps/via/keymap.c:72]:
```suggestion
    [1] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
    [2] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)}
```

File: keyboards/1upkeyboards/1uposu5k/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [8] = LAYOUT_1x5(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [9] = LAYOUT_1x5(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
```
Comment (zvecr) [inline keyboards/1upkeyboards/1uposu5k/keymaps/default/keymap.c:58]:
```suggestion
    )
```

File: keyboards/1upkeyboards/1uposu5k/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
+    [1] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [2] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [3] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [4] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [5] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [6] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [7] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [8] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)},
+    [9] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)}
```
Comment (zvecr) [inline keyboards/1upkeyboards/1uposu5k/keymaps/default/keymap.c:72]:
```suggestion
    [1] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)}
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/1upslider8/config.h [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+#define SLIDER_PIN GP28
+#define MIDI_ADVANCED
+
+#define RGB_DI_PIN GP26
+#define RGB_MATRIX_LED_COUNT 8
+#define RGB_MATRIX_KEYPRESSES // reacts to keypresses
+#define RGBLIGHT_LIMIT_VAL 255
+#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 255
```
Comment (zvecr) [inline keyboards/1upkeyboards/1upslider8/config.h:35]:
```suggestion
```

File: keyboards/1upkeyboards/1uposu5k/config.h [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+#pragma once
+
+#define RGB_DI_PIN GP24
+#define RGB_MATRIX_LED_COUNT 32
+#define RGB_MATRIX_KEYPRESSES // reacts to keypresses
+#define RGBLIGHT_LIMIT_VAL 255
+
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
+#    define RGB_MATRIX_MAXIMUM_BRIGHTNESS 255
```
Comment (zvecr) [inline keyboards/1upkeyboards/1uposu5k/config.h:25]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/1uposu5k/config.h [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define RGB_DI_PIN GP24
```
Comment (drashna) [inline keyboards/1upkeyboards/1uposu5k/config.h:19]: Should be moved to info.json.  and as `ws2812.pin`
```suggestion
```

File: keyboards/1upkeyboards/1uposu5k/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+        KC_Z,    KC_X, LT(1, KC_ESC),  KC_C,    KC_V
+    ),
+
+    [1] = LAYOUT_1x5(
+        RGB_TOG, RGB_MOD, KC_TRNS, RGB_VAD, RGB_VAI
+    )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (drashna) [inline keyboards/1upkeyboards/1uposu5k/keymaps/default/keymap.c:30]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/1upkeyboards/1uposu5k/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+        KC_Z,    KC_X, LT(1, KC_ESC),  KC_C,    KC_V
+    ),
+
+    [1] = LAYOUT_1x5(
+        RGB_TOG, RGB_MOD, KC_TRNS, RGB_VAD, RGB_VAI
+    )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (drashna) [inline keyboards/1upkeyboards/1uposu5k/keymaps/via/keymap.c:30]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/1upkeyboards/1upslider8/config.h [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define OLED_DISPLAY_128X32
```
Comment (drashna) [inline keyboards/1upkeyboards/1upslider8/config.h:19]: Technically isn't needed, as this doesn't do anything, and the default fills this slot.

File: keyboards/1upkeyboards/1upslider8/config.h [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+#define I2C1_SCL_PIN        GP11
+#define I2C1_SDA_PIN        GP10
+#define I2C_DRIVER I2CD1
+#define OLED_BRIGHTNESS 128
+#define OLED_FONT_H "keyboards/1upkeyboards/1upslider8/lib/glcdfont.c"
```
Comment (drashna) [inline keyboards/1upkeyboards/1upslider8/config.h:24]: Might be better to use the default font, and move the image to a "raw" render.

File: keyboards/1upkeyboards/1upslider8/config.h [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+
+
+#define SLIDER_PIN GP28
+#define MIDI_ADVANCED
+
+#define RGB_DI_PIN GP26
```
Comment (drashna) [inline keyboards/1upkeyboards/1upslider8/config.h:30]: Should be moved to info.json, and as `ws2812.pin` (or as `WS2812_DI_PIN` here)
```suggestion
```

File: keyboards/1upkeyboards/1upslider8/keymaps/default/keymap.c [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+
+uint8_t divisor = 0;
+void slider(void){
+  if (divisor++) { // only run the slider function 1/256 times it's called
+      return;
+  }
+  midi_send_cc(&midi_device, 2, 0x3E, 0x7F - (analogReadPin(SLIDER_PIN) >> 3));
+}
+
+void matrix_scan_user(void) {
```
Comment (drashna) [inline keyboards/1upkeyboards/1upslider8/keymaps/default/keymap.c:43]:
```suggestion
void housekeeping_task_user(void) {
```

File: keyboards/1upkeyboards/1upslider8/keymaps/via/keymap.c [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+uint8_t divisor = 0;
+void slider(void){
+  if (divisor++) { // only run the slider function 1/256 times it's called
+      return;
+  }
+  midi_send_cc(&midi_device, 2, 0x3E, 0x7F - (analogReadPin(SLIDER_PIN) >> 3));
+  uprintf("%d string", analogReadPin(SLIDER_PIN) );
+}
+
+void matrix_scan_user(void) {
```
Comment (drashna) [inline keyboards/1upkeyboards/1upslider8/keymaps/via/keymap.c:44]:
```suggestion
void housekeeping_task_user(void) {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by zvecr** [comment]
From the PR checklist:
> * PRs should contain the smallest amount of modifications required for a single change to the codebase
>   * multiple keyboards at the same time is not acceptable

[4658292](https://github.com/qmk/qmk_firmware/pull/21039/commits/46582926d6b8d5084d2f53b150b5589fe13b0c9c) introduces yet another keyboard, which given how many suggestions have already been made to the PR, should not happen.

File: keyboards/1upkeyboards/1uposu5k/keymaps/default/keymap.c (hunk 14 above)
Comment (ziptyze) [reply keyboards/1upkeyboards/1uposu5k/keymaps/default/keymap.c]: done

File: keyboards/1upkeyboards/1uposu5k/keymaps/via/keymap.c (hunk 15 above)
Comment (ziptyze) [reply keyboards/1upkeyboards/1uposu5k/keymaps/via/keymap.c]: done

File: keyboards/1upkeyboards/1upslider8/config.h (hunk 16 above)
Comment (ziptyze) [reply keyboards/1upkeyboards/1upslider8/config.h]: removed for later PR

File: keyboards/1upkeyboards/1upslider8/config.h (hunk 17 above)
Comment (ziptyze) [reply keyboards/1upkeyboards/1upslider8/config.h]: removed for later PR

File: keyboards/1upkeyboards/1upslider8/config.h (hunk 18 above)
Comment (ziptyze) [reply keyboards/1upkeyboards/1upslider8/config.h]: removed for later PR

File: keyboards/1upkeyboards/1upslider8/keymaps/default/keymap.c (hunk 19 above)
Comment (ziptyze) [reply keyboards/1upkeyboards/1upslider8/keymaps/default/keymap.c]: removed for later PR

File: keyboards/1upkeyboards/1upslider8/keymaps/via/keymap.c (hunk 20 above)
Comment (ziptyze) [reply keyboards/1upkeyboards/1upslider8/keymaps/via/keymap.c]: removed for later PR

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**